model/CSVDialog.cc

```diff
@@ -138,7 +140,25 @@ std::string CSVDialog::loadWebFile(const std::string& url)
   const std::string tempStr    = temp.string();
           
   std::ofstream outFile(tempStr, std::ofstream::out);  
-  outFile << res.get().body();                                            
+ 
+  // Deal with zipped csv files 
+  std::string targetStr = target.str();
+  std::string fileExt = ".zip";
+  std::size_t found = targetStr.find(fileExt);
```

Comment:
Can we write the code in such a way as to not assume a specific extension? Someone may want to use a .gz for example, or (less likely) a .z or .Z, which are obsolete zip formats that zlib handles.